### PR TITLE
Small mode selector enhancements

### PR DESCRIPTION
**Pull request by daniel-heppner-ibigroup**
This PR does two things:
- Disables the hover element if there are no sub settings to be displayed.
- Uses flexbox to fix weird wrapping that was happening on long mode settings

### CODE REVIEW

File: packages/trip-form/src/MetroModeSelector/SubSettingsPane.tsx [hunk 1]
```diff
@@ -47,14 +47,20 @@ const SettingsPanel = styled.fieldset`
...
+  display: ${props => (props.flexbox ? "flex" : "box")};
   margin-left: 4px;
   input {
     vertical-align: middle;
   }
 `;
 
 const FormLabelIconWrapper = styled.span`
+  display: flex;
+  align-items: center;
```
Comment (binh-dam-ibigroup) [inline packages/trip-form/src/MetroModeSelector/SubSettingsPane.tsx:62]: Sort CSS.

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM

File: packages/trip-form/src/MetroModeSelector/SubSettingsPane.tsx [hunk 2]
```diff
@@ -47,14 +47,20 @@ const SettingsPanel = styled.fieldset`
   }
 `;
 
-export const SubSettingsCheckbox = styled(CheckboxSelector)`
+export const SubSettingsCheckbox = styled(CheckboxSelector)<{
+  flexbox: boolean;
+}>`
+  display: ${props => (props.flexbox ? "flex" : "box")};
```
Comment (miles-grant-ibigroup) [inline packages/trip-form/src/MetroModeSelector/SubSettingsPane.tsx:53]: What is `display: box`?

File: packages/trip-form/src/MetroModeSelector/MetroModeSelector.story.tsx [hunk 3]
```diff
@@ -69,19 +69,12 @@ const defaultModeButtonDefinitions = [
...
 const modeSettingDefinitionsWithDropdown = [
   ...modeSettingDefinitions,
-  {
-    applicableMode: "TRANSIT",
-    default: "blue",
-    key: "busColor",
-    label: "Bus Color",
-    options: [{ value: "blue", text: "Blue" }],
-    type: "DROPDOWN"
-  },
```
Comment (miles-grant-ibigroup) [inline packages/trip-form/src/MetroModeSelector/MetroModeSelector.story.tsx:79]: Can we keep the dropdown? I know it's broken but it's nice to have as a reference. I'd like to bring it back as an option to replace the sliders at some point

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Working, just a few questions

File: packages/trip-form/src/MetroModeSelector/SubSettingsPane.tsx (hunk 2 above)
Comment (daniel-heppner-ibigroup) [reply packages/trip-form/src/MetroModeSelector/SubSettingsPane.tsx:53]: I have no idea

File: packages/trip-form/src/MetroModeSelector/SubSettingsPane.tsx [hunk 4]
```diff
@@ -47,14 +47,20 @@ const SettingsPanel = styled.fieldset`
   }
 `;
 
-export const SubSettingsCheckbox = styled(CheckboxSelector)`
+export const SubSettingsCheckbox = styled(CheckboxSelector)<{
+  flexbox: boolean;
+}>`
+  display: ${props => (props.flexbox ? "flex" : "block")};
```
Comment (miles-grant-ibigroup) [inline packages/trip-form/src/MetroModeSelector/SubSettingsPane.tsx:53]: Are you sure you want `block` not `inherit`?

File: packages/trip-form/src/MetroModeSelector/SubSettingsPane.tsx [hunk 5]
```diff
@@ -81,7 +87,8 @@ const ModeSettingRenderer = ({
   const labelWithIcon =
     "icon" in setting ? (
       <FormLabelIconWrapper>
-        <span role="none">{setting.icon}</span> {label}
+        <div role="none">{setting.icon}</div>
+        <div> {label}</div>
```
Comment (miles-grant-ibigroup) [inline packages/trip-form/src/MetroModeSelector/SubSettingsPane.tsx:91]: I generally prefer `1ch` of left padding so it's not selectable

**Review by miles-grant-ibigroup — APPROVED** [review]
Great stuff!

File: packages/trip-form/src/MetroModeSelector/SubSettingsPane.tsx (hunk 5 above)
Comment (daniel-heppner-ibigroup) [reply packages/trip-form/src/MetroModeSelector/SubSettingsPane.tsx:91]: I'll use flexbox gap

**Pull request merged**